### DISCUSSION TITLE
Running vacuum with pg_cron extension

### DISCUSSION THREAD

**Discussion by saint-james-fr** [supabase/supabase#30054] [Questions] [answered]
## Context
I need to run maintenance tasks (particularly VACUUM operations) on our database during off-peak hours. These operations typically take longer than the standard postgres role timeout of 2 seconds.

## Question
Are scheduled database maintenance tasks executed via `supabase_cron` role subject to the same statement timeout limitations (2 seconds [edited: 2minutes]) as the standard postgres role? 

## Additional Details
- Current setup: Planning to run VACUUM operations via cron jobs during night hours
- Concern: VACUUM operations typically exceed the 2-second [edited: 2minutes] timeout limit
- Environment: Supabase hosted database

## What I've Tried
I've reviewed the documentation but couldn't find specific information about timeout settings for the `supabase_cron` role.

**Comment by GaryAustin1** [comment]
You say 2 seconds in two places.  The global timeout is 2 minutes.  https://supabase.com/docs/guides/database/postgres/timeouts
API users have a lower timeout set.  

Postgres is the role running cron by default I believe.  I don't see a supabase_cron role at all in my instances, where did you get this from?

**Comment by saint-james-fr** [comment]
Yeah sorry i meant 2 minutes. I did not say there was a supabase_cron role, I was indeed asking what was the role used by pg_cron.
So basically I should just increase the timeout for these kind of vacuum tasks?

**Reply by GaryAustin1** [reply, marked as answer]
Yes.  The link shows several ways to increase the timeout beyond the 2 minutes for tasks inside of the database (or using the direct connection).
